distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java

```diff
@@ -0,0 +1,158 @@
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test.mock.csw;
+
+import static java.lang.Thread.sleep;
+import static com.xebialabs.restito.semantics.Action.composite;
+import static com.xebialabs.restito.semantics.Action.contentType;
+import static com.xebialabs.restito.semantics.Action.custom;
+import static com.xebialabs.restito.semantics.Action.header;
+
+import java.io.IOException;
+import java.time.Duration;
+
+import org.codice.ddf.spatial.ogc.wfs.v2_0_0.catalog.common.Wfs20Constants;
+import org.glassfish.grizzly.http.server.Response;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import com.xebialabs.restito.semantics.Action;
+import com.xebialabs.restito.semantics.Function;
+
+/**
+ * Encompasses the response for a product retrieval for a Restito stub server.
+ * Allows for the simple setup of specific responses, slow sending, and simulated
+ * network failure retry testing.
+ */
+public class ChunkedContent implements Function<Response, Response> {
+    private static final XLogger LOGGER =
+            new XLogger(LoggerFactory.getLogger(ChunkedContent.class));
+
+    private String responseMessage;
+
+    private long messageDelayMs;
+
+    private int numberOfFailures;
+
+    private int numberOfRetries = 0;
+
+    /**
+     * Constructor for a response that has a delay and a number of planned failure.
+     *
+     * @param responseMessage  Message to be sent in response.
+     * @param messageDelay     Time to wait between sending each character of the message.
+     * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     */
+    private ChunkedContent(String responseMessage, Duration messageDelay, int numberOfFailures) {
+        this.responseMessage = responseMessage;
+        this.messageDelayMs = messageDelay.toMillis();
+        this.numberOfFailures = numberOfFailures;
+    }
+
+    /**
+     * Reset states. This will reset any variables used for tracking failure attempts.
+     * This should be called after any complete failure tests to ensure that the response will not
+     * continue waiting for another retry attempt from the calling server.
+     */
+    public void resetFailState() {
```

Comment:
Don't think this can be used anymore now that we're using static factory methods that return `Action` objects. We should remove it.